soc/arm64/xenvm/mmu_regions.c

```diff
@@ -18,6 +19,16 @@ static const struct arm_mmu_region mmu_regions[] = {
 			      DT_REG_ADDR_BY_IDX(DT_INST(0, arm_gic), 1),
 			      DT_REG_SIZE_BY_IDX(DT_INST(0, arm_gic), 1),
 			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_NS),
+
+#ifdef CONFIG_UART_PL011
+
+	MMU_REGION_FLAT_ENTRY("UART0",
+			      DT_REG_ADDR(DT_NODELABEL(uart0)),
+			      DT_REG_SIZE(DT_NODELABEL(uart0)),
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_NS),
+
+#endif
```

Comment:
Uhm, not a big fan of this TBH. I know this is the easiest way but I think we should start rejecting this kind of patches before it is too late.

If you want to do that, please use the proper MMIO device APIs as in `include/zephyr/sys/device_mmio.h` by modifying the serial driver so that it's the driver taking care of MMIO mapping its address space.

Reply:
ok thanks for your review, I will see what I can do

Reply:
I'm going to close this PR and open another one with the pl011 driver modified, thanks